velox/common/file/File.h

```diff
@@ -110,8 +111,16 @@ class ReadFile {
     bytesRead_ = 0;
   }
 
+  // The lifetime of the ReadFile object need not be the same as that of the
```

Comment:
This is very clear. Why is the lifetime of the File object exceeds the lifetime of the MemoryPool? How would the caller know how long the memory pool specified here should be alive? Would it not make sense to pass MemoryPool to openFileForRead method?

Reply:
`openFileForRead` has the same problem unless we pass a `shared_ptr` for the Memory Pool, which is not the right pattern.
The perfect solution would be to pass it to the `pread` call similar to how we currently pass the `Arena`. `std::string_view pread(uint64_t offset, uint64_t length, Arena* arena)`. But that would mean modifying a lot more.  The current solution is what I came up with, which has minimal code changes. What do you think?

Reply:
`preadv` has value especially for AWS S3 as it will avoid multiple small reads. I will look into passing `Arena*` to preadv.

Reply:
Arena does not allocate from a `pool_`.  I guess we should extend Arena as well to allocate from a pool_.
``
void addChunk(int64_t bytes) {
    const int64_t chunkSize = std::max(bytes, kMinChunkSize);
    chunks_.emplace_back(new char[chunkSize]);
    pos_ = chunks_.back().get();
    chunkEnd_ = pos_ + chunkSize;
    reserveEnd_ = pos_ + bytes;
  }
``

Reply:
I don't see Arena used anywhere outside of tests. For large allocations, we use MappedMemory. For smaller allocations - MemoryPool. I wonder whether allocations here fall into large or smaller category. Assume these would be large, hence, MappedMemory would be appropriate to use.

CC: @zzhao0 @oerling

Reply:
> new char[chunkSize]

This code in addChunk also suggests that memory is not accounted anywhere...

Reply:
The scope of using a `pool_` is now larger. That work deserves a separate PR. Can we merge this for now using the previous string-based allocation?